core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -49,7 +50,7 @@ import scala.util.control.{ControlThrowable, NonFatal}
  *   Acceptor has N Processor threads that each have their own selector and read requests from sockets
  *   M Handler threads that handle requests and produce responses back to the processor threads for writing.
  */
-class SocketServer(val config: KafkaConfig, val metrics: Metrics, val time: Time) extends Logging with KafkaMetricsGroup {
+class SocketServer(val config: KafkaConfig, val metrics: Metrics, val time: Time, val credentialCache: Option[ScramCredentialCache] = None) extends Logging with KafkaMetricsGroup {
```

Comment:
It's a bit odd to have a `ScramCredentialsCache` in `SocketServer` and other generic classes (like `ConfigHandler`. It would be nice if we could abstract it so that these classes don't know about specific authentication mechanisms.